core/templates/services/csrf-token.service.ts

```diff
@@ -22,23 +22,35 @@
 
 import { downgradeInjectable } from '@angular/upgrade/static';
 import { Injectable } from '@angular/core';
+// eslint-disable-next-line oppia/disallow-httpclient
+import { HttpClient, HttpBackend } from '@angular/common/http';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CsrfTokenService {
+  // 'tokenPromise' will be null when token is not initialized.
   tokenPromise: PromiseLike<string> | null = null;
+  http: HttpClient;
+
+  constructor(httpBackend: HttpBackend) {
+    this.http = new HttpClient(httpBackend);
```

Comment:
Please add a comment on why `httpBackend` is passed to the constructor.

To bypass the default REQUEST_INTERCEPTER we use which depends on this service. So, to avoid circular dependency.